fair/src/autogluon/fair/utils/max_diff.py

```diff
@@ -0,0 +1,46 @@
+import per_group
+
+def constructor(per_group_metric):
+    def metric(y_true,y_pred,groups):
+        tmp=per_group_metric(y_true,y_pred,groups).max(-1)-per_group_metric(y_true,y_pred,groups).min(-1)
+        assert(tmp>=0)
+        return tmp
+    return metric
+
+##Classification per_group
+expected_prediction = constructor(per_group.expected_prediction)
```

Comment:
Why are all of the files, max_diff, max_ratio, etc. so redundant?  Please merge into a single class with the all constructors here.